test/e2e/specs/plugins-api.test.js

```diff
@@ -75,4 +77,28 @@ describe( 'Using Plugins API', () => {
 			expect( pluginSidebarClosed ).toBeNull();
 		} );
 	} );
+
+	describe( 'Annotations', () => {
+		beforeAll( () => {
+			observeFocusLoss();
+		} );
+
+		it( 'Allows a block to be anotated', async () => {
+			await newPost();
+
+			await page.keyboard.type( 'Title' + '\n' + 'Paragraph to annotate' );
+			await clickOnMoreMenuItem( 'Sidebar title plugin' );
+
+			let annotations = await page.$$( ANNOTATIONS_SELECTOR );
+			expect( annotations ).toHaveLength( 0 );
+
+			// Click add annotation button.
+			const addAnnotationButton = ( await page.$x( "//button[contains(text(), 'Add annotation')]" ) )[ 0 ];
+			// console.log( addAnnotationButton );
+			await addAnnotationButton.click();
+
+			annotations = await page.$$( ANNOTATIONS_SELECTOR );
+			expect( annotations ).toHaveLength( 1 );
```

Comment:
Could we test adding a simple annotation to a paragraph block? So checking if the annotation is saved, displayed in the editor, and stripped in HTML view for example.

Reply:
Added this in https://github.com/WordPress/gutenberg/pull/7718/commits/844c4a6053a63c0040e252949cf1c9a542e17224.

Reply:
I would like to see more. I don't see how someone would add an annotation to the current selection.